fields/types/sortableRelateds/SortableRelateds.js

```diff
@@ -0,0 +1,298 @@
+        if (fieldName === ':_id' && Keystone.item) {
+          filters[key] = Keystone.item.id;
+          return;
+        }
+      } else {
+        filters[key] = value;
+      }
+    }, this);
+
+    var parts = [];
+    _.each(filters, function(val, key) {
+      parts.push('filters[' + key + '][value]=' + encodeURIComponent(val));
+    });
+    return parts.join('&');
+  },
+
+  cacheItem(item) {
+    item.href = Keystone.adminPath + '/' + this.props.refList.path + '/' + item.id;
+    this._itemsCache[item.id] = item;
+  },
+
+  loadPostInfo(postIds) {
+    if (!postIds) {
+      this.setState({ loading: false, value: null });
+      return;
+    };
+    postIds = Array.isArray(postIds) ? postIds : postIds.split(',');
+    let cachedValues = postIds.map(id => this._itemsCache[id]).filter(i => i);
+    if (cachedValues.length === postIds.length) {
+      this.setState({ loading: false, value: this.props.many ? cachedValues : cachedValues[0] });
+      return;
+    }
+    this.setState({ loading: true, value: null });
+    async.map(postIds, (postId, done) => {
+      xhr({
+        url: Keystone.adminPath + '/api/' + this.props.refList.path + '/' + postId,
+        responseType: 'json',
+      }, (err, resp, data) => {
+        if (err || !data) return done(err);
+        this.cacheItem(data);
+        done(err, data);
+      });
+    }, (err, expanded) => {
+      if (!this.isMounted()) return;
+      this.setState({ loading: false, value: this.props.many ? expanded : expanded[0] });
+    });
+  },
+
+  loadOptions(postIds) {
+    const filters = this.buildFilters();
+    xhr({
+      url: Keystone.adminPath + '/api/' + this.props.refList.path + '?basic&search=' + '' + '&' + filters,
+      responseType: 'json',
+    }, (err, resp, data) => {
+      if (err) {
+        console.error('Error loading items:', err);
+        return;
+      }
+      data.results.forEach(post => {
+        if (post) {
+          this._options.push({ label: post.slug, value: post.id });
+          this.cacheItem(post);
+        }
+      });
+      postIds = Array.isArray(postIds) ? postIds : postIds.split(',');
+      this.setState({ options: this._options.filter(postOption => postOption && !postIds.includes(postOption.value)) });
+    });
+  },
+
+  updatePickUpStatus() {
+    const { value } = this.state;
+    if (!Array.isArray(value)) {
+      return;
+    }
+
+    let numPickedUp = 0;
+    value.forEach(post => {
+      if (post && post.isPickedUpToRemove) {
+        numPickedUp++;
+      }
+    });
+
+    let pickUpStatus = PickUp.INDETERMINATE;
+    if (numPickedUp === 0) {
+      pickUpStatus = PickUp.NONE;
+    } else if (numPickedUp === value.length) {
+      pickUpStatus = PickUp.ALL;
+    }
+
+    this.setState({ pickUpStatus: pickUpStatus });
+  },
+
+  onPickUpSingle(postId) {
+    const { value } = this.state;
+    const post = Array.isArray(value) ? value.find(post => post && post.id === postId) : undefined;
```

Comment:
Or early return if the value is not array ?

if (!Array.isArray(value)) return
const post = value.find(...)

Reply:
refined.